### PR TITLE
Generate instrumentation packages setup.py files

### PR DESCRIPTION
**Pull request by owais**
# Description

All instrumentations packages have almost exactly same setup.py files. This commit adds a python script that generates it from a source template. This dramatically reduces the time and effort required to update all instrumentation setup.py files, and also reduces chances of making manual mistakes.

I'm working on another feature where I needed to update setup.py files for all insrumentations. I'll use this once it is merged.

Another alternative is to take out common code as yet another package and add it as a `setup_requires` dependency so it is installed in target envs before setup.py runs but this adds another runtime dependency for users and doesn't give them any benefits. It's also not heavily used in the eco-system so non standard tools (other than pip) may not work well with it.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Tooling enhancement

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-wsgi/setup.py [hunk 1]
```diff
@@ -11,6 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+
+# DO NOT EDIT. THIS FILE WAS AUTOGENERATED FROM templaes/wsgi.
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-wsgi/setup.py:16]: looks like a typo sneaked into the generated code, though it doesnt appear in the generator 🤔

File: instrumentation/opentelemetry-instrumentation-urllib3/setup.py [hunk 2]
```diff
@@ -11,6 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+
+# DO NOT EDIT. THIS FILE WAS AUTOGENERATED FROM templaes/urllib3.
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-urllib3/setup.py:16]: should this say it was generated from instrumentation_setup.py?

File: instrumentation/opentelemetry-instrumentation-wsgi/setup.py [hunk 3]
```diff
@@ -11,6 +11,12 @@
...
+# Run `python scripts/generate_setuppy.py` to regenerate.
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-wsgi/setup.py:17]: Nit: I see why you named this the way you did, but I would rename the script `generate_setup.py`, it's easier to read.

**Review by codeboten — COMMENTED** [review]
Would love to see this extended to include the version file at some point as well, though I don't know if the instrumentation versions will always been the same or not.

File: instrumentation/opentelemetry-instrumentation-wsgi/setup.py (hunk 1 above)
Comment (owais) [reply instrumentation/opentelemetry-instrumentation-wsgi/setup.py:16]: Yea, looks like I updated the generator but didn't run it before checking in.

File: instrumentation/opentelemetry-instrumentation-urllib3/setup.py (hunk 2 above)
Comment (owais) [reply instrumentation/opentelemetry-instrumentation-urllib3/setup.py:16]: Yes. This is incorrect. Fixing.

**Review by codeboten — APPROVED** [review]
Nice, this really does cleanup things and ensures consistency in the instrumentations.

**Pull request merged**